File: src/code-samples/langsmith/smithdb-migration/traces-list-runs-basic-before.go

// :snippet-start: traces-list-runs-basic-before-go
// :codegroup-tab: Before
package main

import (
	"context"
	"fmt"

	"github.com/langchain-ai/langsmith-go"
)

func main() {
	ctx := context.Background()
	client := langsmith.NewClient()

	sessions, err := client.Sessions.List(ctx, langsmith.SessionListParams{
		Name:  langsmith.F("default"),
		Limit: langsmith.F(int64(1)),
	})
	if err != nil {
		panic(err.Error())
	}
	projectID := sessions.Items[0].ID
	traceID := "<trace-id>"
	// :remove-start:
	rootRuns, err := client.Runs.Query(ctx, langsmith.RunQueryParams{
		Session: langsmith.F([]string{projectID}),
		IsRoot:  langsmith.F(true),
		Limit:   langsmith.F(int64(1)),
	})
	if err != nil {
		panic(err.Error())
	}
	traceID = rootRuns.Runs[0].TraceID
	// :remove-end:

	runs, err := client.Runs.Query(ctx, langsmith.RunQueryParams{
		Session: langsmith.F([]string{projectID}),
		Trace:   langsmith.F(traceID),
	})
	if err != nil {
		panic(err.Error())
	}
	for _, run := range runs.Runs {
		fmt.Println(run.Name, run.RunType, run.Status)
	}
}
// :snippet-end:

File: src/code-samples/langsmith/smithdb-migration/traces-list-runs-basic-before.ts

// :snippet-start: traces-list-runs-basic-before-js
// :codegroup-tab: Before
import { Client } from "langsmith";

const client = new Client();
const project = await client.readProject({ projectName: "default" });
let traceId = "<trace-id>";
// :remove-start:
for await (const run of client.listRuns({ projectId: project.id, isRoot: true, limit: 1 })) {
  traceId = run.trace_id!;
  break;
}
// :remove-end:
const runs = [];
for await (const run of client.listRuns({ projectId: project.id, traceId })) {
  runs.push(run);
}
for (const run of runs) {
  console.log(run.name, run.run_type, run.status);
}
// :snippet-end:
